### PR TITLE
Add swagger authorization header support

### PR DESCRIPTION
**Pull request by dongxiaoman**
## Description
This PR adds the `Authorization: xxx` header support into the Swagger UI in Pinot.
Now when the users opens Pinot Swagger, they will see an extra "Authorize" button.
1. If the user does not authorize, the Swagger will behave like before, send all the API requests through
2. If the user clicks "Authorize" button, the user can pass in one extra bearer token and send them into Pinot

Below is a gif for authorization (**Note that the Authorization is not mandatory, it just plainly adds a header**)
authorization

## Release Notes
- Added Authentication header option into Swagger UI for all endpoints

## Tests Done
Run the `PinotQuickStart` example and recorded the screen like above shows

### CODE REVIEW

**Comment by dongxiaoman** [comment]
cc @apucher @xiangfu0

File: pinot-spi/src/main/java/org/apache/pinot/spi/utils/CommonConstants.java [hunk 1]
```diff
@@ -41,6 +41,7 @@ private CommonConstants() {
   public static final String DEFAULT_METRICS_FACTORY_CLASS_NAME =
       "org.apache.pinot.plugin.metrics.yammer.YammerMetricsFactory";
 
+  public static final String DEFAULT_SWAGGER_AUTHORIZATION_VALUE = "oauth";
```
Comment (apucher) [inline pinot-spi/src/main/java/org/apache/pinot/spi/utils/CommonConstants.java:44]: maybe change the name to `SWAGGER_AUTHORIZATION_KEY` - and reuse it in both, the `@ApiKeyAuthDefinition` and the `@Authorization` annotations rather then the hardcoded `"oauth"` string?

Comment (dongxiaoman) [reply pinot-spi/src/main/java/org/apache/pinot/spi/utils/CommonConstants.java]: good point. Changed now

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotControllerHealthCheck.java [hunk 2]
```diff
@@ -37,8 +38,10 @@
 import org.apache.pinot.controller.BaseControllerStarter;
 import org.apache.pinot.controller.ControllerConf;
 
+import static org.apache.pinot.spi.utils.CommonConstants.SWAGGER_AUTHORIZATION_KEY;
 
-@Api(tags = Constants.HEALTH_TAG)
+
+@Api(tags = Constants.HEALTH_TAG, authorizations = {@Authorization(value = SWAGGER_AUTHORIZATION_KEY)})
 @Path("/")
```
Comment (apucher) [inline pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotControllerHealthCheck.java:45]: should there be additional swagger annotations as everywhere else?

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotLeadControllerRestletResource.java [hunk 3]
```diff
@@ -83,8 +92,8 @@ public LeadControllerResponse getLeadersForAllTables() {
       String partitionName = LeadControllerUtils.generatePartitionName(partitionId);
       String participantInstanceId =
           getParticipantInstanceIdFromExternalView(leadControllerResourceExternalView, partitionName);
-      leadControllerEntryMap
-          .putIfAbsent(partitionName, new LeadControllerEntry(participantInstanceId, new ArrayList<>()));
+      leadControllerEntryMap.putIfAbsent(partitionName,
+          new LeadControllerEntry(participantInstanceId, new ArrayList<>()));
```
Comment (apucher) [inline pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotLeadControllerRestletResource.java:96]: seems like an unrelated formatting change. I'd revert that

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTableTenantConfigs.java [hunk 4]
```diff
@@ -58,9 +66,9 @@ public class PinotTableTenantConfigs {
   @Path("/tables/{tableName}/rebuildBrokerResourceFromHelixTags")
   @ApiOperation(value = "Rebuild broker resource for table", notes = "when new brokers are added")
   @ApiResponses(value = {
-      @ApiResponse(code = 200, message = "Success"),
-      @ApiResponse(code = 400, message = "Bad request: table name has to be with table type"),
-      @ApiResponse(code = 500, message = "Internal error rebuilding broker resource or serializing response")
+      @ApiResponse(code = 200, message = "Success"), @ApiResponse(code = 400, message = "Bad request: table name has "
+      + "to be with table type"), @ApiResponse(code = 500, message = "Internal error rebuilding broker resource or "
+      + "serializing response")
```
Comment (apucher) [inline pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTableTenantConfigs.java:71]: unrelated formatting

File: pinot-server/src/main/java/org/apache/pinot/server/api/resources/MmapDebugResource.java [hunk 5]
```diff
@@ -65,11 +75,10 @@ public List<String> getOffHeapSizes() {
...
   @Produces(MediaType.APPLICATION_JSON)
   @Path("/memory/offheap/table/{tableName}")
-  @ApiOperation(value = "Show off heap memory consumed by latest mutable segment",
-      notes = "Returns off heap memory consumed by latest consuming segment of realtime table")
+  @ApiOperation(value = "Show off heap memory consumed by latest mutable segment", notes = "Returns off heap memory "
+      + "consumed by latest consuming segment of realtime table")
   @ApiResponses(value = {
-      @ApiResponse(code = 200, message = "Success"),
-      @ApiResponse(code = 500, message = "Internal server error"),
+      @ApiResponse(code = 200, message = "Success"), @ApiResponse(code = 500, message = "Internal server error"),
```
Comment (apucher) [inline pinot-server/src/main/java/org/apache/pinot/server/api/resources/MmapDebugResource.java:81]: should this have addl swagger annotations? also, unrelated formatting

File: pinot-tools/src/main/java/org/apache/pinot/tools/service/api/resources/PinotServiceManagerInstanceResource.java [hunk 6]
```diff
@@ -70,8 +78,7 @@ public class PinotServiceManagerInstanceResource {
   @Path("/instances")
   @ApiOperation(value = "Get Pinot Instances Status")
   @ApiResponses(value = {
-      @ApiResponse(code = 200, message = "Instance Status"),
-      @ApiResponse(code = 500, message = "Internal server error")
+      @ApiResponse(code = 200, message = "Instance Status"), @ApiResponse(code = 500, message = "Internal server error")
```
Comment (apucher) [inline pinot-tools/src/main/java/org/apache/pinot/tools/service/api/resources/PinotServiceManagerInstanceResource.java:81]: unrelated formatting here, and below

**Review by apucher — APPROVED** [review]

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotControllerHealthCheck.java (hunk 2 above)
Comment (dongxiaoman) [reply pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotControllerHealthCheck.java:45]: Yes good catch. I was thinking maybe health check does not need authz but clearly this is for Swagger so it does not matter

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotLeadControllerRestletResource.java (hunk 3 above)
Comment (dongxiaoman) [reply pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotLeadControllerRestletResource.java]: Seems my IDE's style format is never set correct. Need to double check that again.

File: pinot-server/src/main/java/org/apache/pinot/server/api/resources/MmapDebugResource.java (hunk 5 above)
Comment (dongxiaoman) [reply pinot-server/src/main/java/org/apache/pinot/server/api/resources/MmapDebugResource.java]: The Swagger annotation is added to the class instead of each endpoints.
 The format change is good catch.

**Review by xiangfu0 — APPROVED** [review]
